antarest/study/storage/rawstudy/model/filesystem/inode.py

```diff
@@ -148,17 +148,17 @@ def _extract_file_to_tmp_dir(
             The actual path of the extracted file
             the tmp_dir object which MUST be cleared after use of the file
         """
-        if self.config.zip_path is None:
+        if self.config.archive_path is None:
             raise ShouldNotHappenException()
-        inside_zip_path = str(self.config.path)[len(str(self.config.zip_path)[:-4]) + 1 :]
-        if self.config.zip_path:
-            return extract_file_to_tmp_dir(self.config.zip_path, Path(inside_zip_path))
+        inside_archive_path = str(self.config.path)[len(str(self.config.archive_path)[:-4]) + 1 :]
```

Comment:
Even if you didn't write it it's ugly and false (we should use [:-3] if we are in 7z and not -4 which corresponds to zip).
We should change it to:
`inside_archive_path = self.config.path.relative_to(self.config.archive_path.parent / self.config.study_id)` which is reliable and doesn't need to recast it to a path afterwards

Also with this you will never have the `/` at the beginning of the path so you can remove the weird lines I pointed out in my last comment